src/mygrad/tensor_base.py

```diff
@@ -431,7 +451,7 @@ def null_gradients(self, clear_graph=True):
         """
         self._null_gradients(clear_graph=clear_graph, seen=set())
 
-    def _null_gradients(self, *, clear_graph: bool, seen: Set["Tensor"]):
+    def _null_gradients(self, *, clear_graph: bool, seen: Set["Tensor"]) -> None:
```

Comment:
A `Set[Tensor]` could be very large, right?  Not sure it helps here, but in general I'd be looking for places to use various [`weakref` collections](https://docs.python.org/3/library/weakref.html) to avoid keeping them around for longer than necessary.

Reply:
Oh wow, I didn't even know about the `weakref` collections! This will be very very useful. Thank you! ❤️

Reply:
The standard library strikes again!  So many useful data structures and tools, so little advertising...